### PR TITLE
binary_messenger missing cstdint

### PR DESCRIPTION
**Pull request by jwinarske**
This PR adds cstdint to binary_messenger.h.  Without this building plugin support for generic Linux fails.  Not related to the GTK source tree.

*Replace this paragraph with a description of what this PR is changing or adding, and why. Consider including before/after screenshots.*

*List which issues are fixed by this PR. You must list at least one issue.*

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by bdero — APPROVED** [review]
LGTM

File: shell/platform/common/client_wrapper/include/flutter/binary_messenger.h [hunk 1]
```diff
@@ -7,6 +7,8 @@
 
 #include <functional>
 #include <string>
+#include <cstdint>
```
Comment (bdero) [inline shell/platform/common/client_wrapper/include/flutter/binary_messenger.h:10]: I think the linter will complain that this needs to be in alphabetical order

Comment (jwinarske) [reply shell/platform/common/client_wrapper/include/flutter/binary_messenger.h]: good point

**Comment by bdero** [comment]
Huh, it looks like we had the big clang-tidy switcheroo yesterday: https://github.com/flutter/engine/pull/48145

FYI @matanlurey, I thought there was a clang tidy lint to enforce direct stl includes, but searching around [the list](https://clang.llvm.org/extra/clang-tidy/checks/list.html) I'm not locating one...

**Comment by matanlurey** [comment]
> the list

No, we have no such thing. See https://github.com/flutter/flutter/issues/133417, I'd love help doing this :)

**Comment by matanlurey** [comment]
I'm reverting this PR as we didn't agree to land it (see [Discord thread](https://discord.com/channels/608014603317936148/608018585025118217/1179244612167155733)).

**Comment by jwinarske** [comment]
grep in main (pre-PR):
```
common/client_wrapper/include/flutter/texture_registrar.h:#include <cstdint>
common/client_wrapper/include/flutter/encodable_value.h:#include <cstdint>
common/client_wrapper/byte_buffer_streams.h:#include <cstdint>
```

**Comment by jwinarske** [comment]
Originating error:
```
In file included from /home/joel/test/example/plugins/pigeon/messages.g.cc:9:
In file included from /home/joel/test/example/plugins/pigeon/messages.g.h:9:
In file included from /home/joel/test/third_party/flutter/shell/platform/common/client_wrapper/include/flutter/basic_message_channel.h:12:
/home/joel/test/third_party/flutter/shell/platform/common/client_wrapper/include/flutter/binary_messenger.h:16:34: error: unknown type name 'uint8_t'
typedef std::function<void(const uint8_t* reply, size_t reply_size)>
                                 ^
```

**Comment by jwinarske** [comment]
@stuartmorgan What is your actionable feedback for me on this?

Does it need a test case, if so where in the tree should I locate it?

**Pull request merged**